llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -665,101 +780,151 @@ def tensor_hook(
-                    license_filename = _maybe_get_license_filename(
-                        local_save_path,
-                        self.pretrained_model_name,
+                    new_model_instance = self.transform_model_pre_registration(
+                        new_model_instance,
                     )
-                    if license_filename is not None:
-                        mlflow_logger._mlflow_client.log_artifact(
-                            mlflow_logger._run_id,
-                            os.path.join(local_save_path, license_filename),
-                        )
 
-                    self.pre_register_edit(local_save_path,)
+                    new_model_instance.save_pretrained(register_save_dir)
+                    original_tokenizer.save_pretrained(register_save_dir)
 
-                    # Save the monitor process to be restored after registering the model.
-                    if hasattr(mlflow_logger, 'monitor_process'):
-                        monitor_process = mlflow_logger.monitor_process  # type: ignore
-                        mlflow_logger.monitor_process = None  # type: ignore
-                    else:
-                        monitor_process = None
-
-                    # Spawn a new process to register the model.
-                    process = SpawnProcess(
-                        target=_register_model_with_run_id_multiprocess,
-                        kwargs={
-                            'mlflow_logger':
-                                mlflow_logger,
-                            'composer_logging_level':
-                                logging.getLogger('composer').level,
-                            'model_uri':
-                                local_save_path,
-                            'name':
-                                self.mlflow_registered_model_name,
-                            'await_creation_for':
-                                3600,
-                        },
-                    )
-                    process.start()
+                    self.pre_register_edit(register_save_dir)
 
-                    # Restore the monitor process.
-                    if monitor_process is not None:
-                        mlflow_logger.monitor_process = monitor_process  # type: ignore
-                    self.register_processes.append(process)
+                    for mlflow_logger in self.mlflow_loggers:
+                        log.debug(
+                            f'Registering model to UC at {mlflow_logger.model_registry_prefix}.{self.mlflow_registered_model_name}',
+                        )
 
-                    # Save the temporary directory to be cleaned up later.
-                    if use_temp_dir:
-                        self.temp_save_dir = temp_save_dir
+                        # Patch the mlflow save_model function to remove duplicate tokenizer files and log the license file.
+                        _patch_mlflow_save_model(
```

Comment:
did you confirm patch applies in main process -> sub process?

Reply:
checked your experiment and i don't think the save model patch propagated to the subprocess (which makes sense)